Build failure: qgnomeplatform

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

This happens when trying to re-build the latest version of nixos running gnome.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310730084

### Relevant log output

```console
Running phase: qtPreHook
Running phase: unpackPhase
unpacking source archive /nix/store/p3fn20xh3f1zlg3grw6vcrx1nxg5h4i6-source
source root is source
Running phase: patchPhase
applying patch /nix/store/m0fd6mkhbp4cvh1gm8w92iz0l99aq548-hardcode-gsettings.patch
patching file src/common/gnomesettings.cpp
Hunk #1 succeeded at 154 (offset 4 lines).
applying patch /nix/store/cgawmp86v9knqkcvfpj35qzx0w280jkw-qt6-cursor-fix.patch
patching file src/common/gnomesettings.cpp
Hunk #1 succeeded at 218 (offset 8 lines).
Hunk #2 succeeded at 355 (offset 8 lines).
Hunk #3 succeeded at 401 (offset 8 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/share/doc/QGnomePlatform -DCMAKE_INSTALL_INFODIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4 -DGLIB_SCHEMAS_DIR=/nix/store/03iq7iqv9avzxhxpf791n0asqg11rj3a-gsettings-desktop-schemas-48.0/share/gsettings-schemas/gsettings-desktop-schemas-48.0/glib-2.0/schemas -DQT_PLUGINS_DIR=/nix/store/l4ad4crg9lgpx52ihpbyszcayjcdvzp6-qgnomeplatform-0.8.4/lib/qt-6/plugins -DCMAKE_POLICY_VERSION_MINIMUM:STRING=3.31 -DUSE_QT6=true
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/as2rg79vk4z7frb7hy27nmadc3fp1a4j-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
-- Found PkgConfig: /nix/store/0skggkamysbh0ksc3582lb69wzrpabcg-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'glib-2.0'
--   Found glib-2.0, version 2.84.4
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Found GSettingSchemas: TRUE
-- Checking for module 'gtk+-3.0'
--   Found gtk+-3.0, version 3.24.49
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Found Wayland_Client: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-client.so (found version "1.24.0")
-- Found Wayland_Server: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-server.so (found version "1.24.0")
-- Found Wayland_Cursor: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-cursor.so (found version "1.24.0")
-- Found Wayland_Egl: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-egl.so (found version "18.1.0")
-- Found Wayland: /nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-client.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-server.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-cursor.so;/nix/store/sk1nyvzlzffrmk2xmy14rswyafhp7bz8-wayland-1.24.0/lib/libwayland-egl.so (found version "1.24.0")
-- Found WaylandScanner: /nix/store/pd0fbk9yrygb8c5p36j4cazpf6lw8v75-wayland-scanner-1.24.0-bin/bin/wayland-scanner
-- Enabling Qt Wayland decoration shadows support
-- NOTE: This support requires changes in Qt Wayland from Qt 6.2
-- The following OPTIONAL packages have been found:

 * Qt6DBusTools (required version >= 6.10.0)
 * Qt6DBus (required version >= 6.2.0)
 * OpenGL
 * Vulkan
 * WrapVulkanHeaders
 * Qt6CoreTools (required version >= 6.10.0)
 * Qt6GuiTools (required version >= 6.10.0)
 * Qt6WidgetsTools (required version >= 6.10.0)
 * Qt6WaylandScannerTools (required version >= 6.10.0)

-- The following REQUIRED packages have been found:

 * QT
 * Qt6 (required version >= 6.2.0)
 * GSettingSchemas
 * Qt6Core (required version >= 6.2.0)
 * Qt6Gui (required version >= 6.2.0)
 * Qt6Widgets (required version >= 6.2.0)
 * AdwaitaQt6 (required version >= 1.4.1)
 * PkgConfig
 * Qt6WaylandClient (required version >= 6.2.0), Qt Wayland decoration support
   Required for QGnomePlatform decoration plugin

-- Configuring done (4.2s)
CMake Error at src/common/CMakeLists.txt:7 (target_link_libraries):
  Target "qgnomeplatform6" links to:

    Qt6::GuiPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at src/decoration/CMakeLists.txt:12 (target_link_libraries):
  Target "qgnomeplatformdecoration" links to:

    Qt6::GuiPrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025
    QT_PLUGINS_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Additional context

Started happening with the latest release. Yesterdays release worked without issues.

Seems like there was a similar failure resolved recently https://github.com/NixOS/nixpkgs/issues/449595.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.1-zen1, NixOS, 25.11 (Xantusia), 25.11.20251015.544961d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/rcsmdj802sk3w9l8m0b0bvdy0jiw2wgr-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/qgnomeplatform/default.nix
+++ b/pkgs/development/libraries/qgnomeplatform/default.nix
@@ -36,6 +36,9 @@ stdenv.mkDerivation rec {
     # Backport cursor fix for Qt6 apps
     # Adjusted from https://github.com/FedoraQt/QGnomePlatform/pull/138
     ./qt6-cursor-fix.patch
+
+    # fixing build with Qt>=6.10
+    ./qt6_10.patch
   ];
 
   nativeBuildInputs = [

--- a/pkgs/development/libraries/qgnomeplatform/qt6_10.patch
+++ b/pkgs/development/libraries/qgnomeplatform/qt6_10.patch
@@ -1,0 +1,140 @@
+diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
+index e6dcf06..287824c 100644
+--- a/src/common/CMakeLists.txt
++++ b/src/common/CMakeLists.txt
+@@ -3,6 +3,10 @@ set(common_SRCS
+     gnomesettings.cpp
+ )
+ 
++if(QT_VERSION_MAJOR EQUAL 6)
++    find_package(Qt6 REQUIRED COMPONENTS GuiPrivate)
++endif()
++
+ add_library(qgnomeplatform${LIBQGNOMEPLATFORM_SUFFIX} SHARED ${common_SRCS})
+ target_link_libraries(qgnomeplatform${LIBQGNOMEPLATFORM_SUFFIX}
+     Qt${QT_VERSION_MAJOR}::Core
+diff --git a/src/decoration/CMakeLists.txt b/src/decoration/CMakeLists.txt
+index a6fb227..7b6647c 100644
+--- a/src/decoration/CMakeLists.txt
++++ b/src/decoration/CMakeLists.txt
+@@ -8,6 +8,11 @@ set(decoration_SRCS
+     qgnomeplatformdecoration.cpp
+ )
+ 
++if(QT_VERSION_MAJOR EQUAL 6)
++    find_package(Qt6 REQUIRED COMPONENTS GuiPrivate)
++    find_package(Qt6 REQUIRED COMPONENTS WaylandClientPrivate)
++endif()
++
+ add_library(qgnomeplatformdecoration MODULE ${decoration_SRCS})
+ target_link_libraries(qgnomeplatformdecoration
+     qgnomeplatform${LIBQGNOMEPLATFORM_SUFFIX}
+diff --git a/src/decoration/qgnomeplatformdecoration.cpp b/src/decoration/qgnomeplatformdecoration.cpp
+index a719cb3..79c0be0 100644
+--- a/src/decoration/qgnomeplatformdecoration.cpp
++++ b/src/decoration/qgnomeplatformdecoration.cpp
+@@ -616,8 +616,12 @@ void QGnomePlatformDecoration::processMouseTop(QWaylandInputDevice *inputDevice,
+         if (local.x() <= margins().left()) {
+             //top left bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++            waylandWindow()->applyCursor(inputDevice, Qt::SizeFDiagCursor);
++#else
+             waylandWindow()->setMouseCursor(inputDevice, Qt::SizeFDiagCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+             startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_TOP_LEFT, b);
+ #else
+@@ -626,8 +630,12 @@ void QGnomePlatformDecoration::processMouseTop(QWaylandInputDevice *inputDevice,
+         } else if (local.x() > window()->width() + margins().left()) {
+             //top right bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++            waylandWindow()->applyCursor(inputDevice, Qt::SizeBDiagCursor);
++#else
+             waylandWindow()->setMouseCursor(inputDevice, Qt::SizeBDiagCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+             startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_TOP_RIGHT, b);
+ #else
+@@ -636,8 +644,12 @@ void QGnomePlatformDecoration::processMouseTop(QWaylandInputDevice *inputDevice,
+         } else {
+             //top resize bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++            waylandWindow()->applyCursor(inputDevice, Qt::SplitVCursor);
++#else
+             waylandWindow()->setMouseCursor(inputDevice, Qt::SplitVCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+             startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_TOP, b);
+ #else
+@@ -682,8 +694,12 @@ void QGnomePlatformDecoration::processMouseBottom(QWaylandInputDevice *inputDevi
+     if (local.x() <= margins().left()) {
+         //bottom left bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++        waylandWindow()->applyCursor(inputDevice, Qt::SizeBDiagCursor);
++#else
+         waylandWindow()->setMouseCursor(inputDevice, Qt::SizeBDiagCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+         startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_BOTTOM_LEFT, b);
+ #else
+@@ -692,8 +708,12 @@ void QGnomePlatformDecoration::processMouseBottom(QWaylandInputDevice *inputDevi
+     } else if (local.x() > window()->width() + margins().right()) {
+         //bottom right bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++        waylandWindow()->applyCursor(inputDevice, Qt::SizeFDiagCursor);
++#else
+         waylandWindow()->setMouseCursor(inputDevice, Qt::SizeFDiagCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+         startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_BOTTOM_RIGHT, b);
+ #else
+@@ -702,8 +722,12 @@ void QGnomePlatformDecoration::processMouseBottom(QWaylandInputDevice *inputDevi
+     } else {
+         //bottom bit
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++        waylandWindow()->applyCursor(inputDevice, Qt::SplitVCursor);
++#else
+         waylandWindow()->setMouseCursor(inputDevice, Qt::SplitVCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+         startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_BOTTOM, b);
+ #else
+@@ -717,8 +741,12 @@ void QGnomePlatformDecoration::processMouseLeft(QWaylandInputDevice *inputDevice
+     Q_UNUSED(local)
+     Q_UNUSED(mods)
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++    waylandWindow()->applyCursor(inputDevice, Qt::SplitHCursor);
++#else
+     waylandWindow()->setMouseCursor(inputDevice, Qt::SplitHCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+         startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_LEFT, b);
+ #else
+@@ -731,8 +759,12 @@ void QGnomePlatformDecoration::processMouseRight(QWaylandInputDevice *inputDevic
+     Q_UNUSED(local)
+     Q_UNUSED(mods)
+ #if QT_CONFIG(cursor)
++#if (QT_VERSION >= QT_VERSION_CHECK(6, 10, 0))
++    waylandWindow()->applyCursor(inputDevice, Qt::SplitHCursor);
++#else
+     waylandWindow()->setMouseCursor(inputDevice, Qt::SplitHCursor);
+ #endif
++#endif
+ #if (QT_VERSION < QT_VERSION_CHECK(5, 13, 0))
+         startResize(inputDevice, WL_SHELL_SURFACE_RESIZE_RIGHT, b);
+ #else
+

